numpy/distutils/ccompiler_opt.py

```diff
@@ -579,45 +579,38 @@ def dist_test(self, source, flags):
         return test
 
     def dist_info(self):
-        """Return a string containing all environment information, required
-        by the abstract class '_CCompiler' to discovering the platform
-        environment, also used as a cache factor in order to detect
-        any changes from outside.
+        """
+        Return a tuple conatins info about (platform, compiler, extra_args),
+        required by the abstract class '_CCompiler' to discovering the
+        platform environment, also used as a cache factor in order to detect
+        any changes happend from outside.
         """
         if hasattr(self, "_dist_info"):
             return self._dist_info
-        # play it safe
-        cc_info = ""
-        compiler = getattr(self._ccompiler, "compiler", None)
-        if compiler is not None:
-            if isinstance(compiler, str):
-                cc_info += compiler
-            elif hasattr(compiler, "__iter__"):
-                cc_info += ' '.join(compiler)
-        # in case if 'compiler' attribute doesn't provide anything
-        cc_type = getattr(self._ccompiler, "compiler_type", "")
-        if cc_type in ("intelem", "intelemw", "mingw64"):
-            cc_info += "x86_64"
+
+        cc_type = getattr(self._ccompiler, "compiler_type", '')
+        if cc_type in ("intelem", "intelemw"):
+            platform = "x86_64"
         elif cc_type in ("intel", "intelw", "intele"):
-            cc_info += "x86"
-        elif cc_type in ("msvc", "mingw32"):
-            import platform
-            if platform.architecture()[0] == "32bit":
-                cc_info += "x86"
-            else:
-                cc_info += "x86_64"
+            platform = "x86"
         else:
-            # the last hope, too bad for cross-compiling
-            import platform
-            cc_info += platform.machine()
+            from distutils.util import get_platform
+            platform = get_platform()
```

Comment:
according to [distutils.util doc](https://docs.python.org/3/distutils/apiref.html#module-distutils.util), `get_platform()` should be friendly with osx universal binaries.